MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CustomerServiceDefault.java

```diff
@@ -51,26 +53,26 @@
 
     final Scheduler callbackScheduler;
 
-    CustomerToken customerToken;
+    final AtomicReference<CustomerToken> customerTokenRef = new AtomicReference<>();
 
     CustomerServiceDefault(
             final Retrofit retrofit,
             final CustomerToken customerToken,
             final Scheduler callbackScheduler
     ) {
         this.retrofitService = retrofit.create(CustomerRetrofitService.class);
-        this.customerToken = customerToken;
+        this.customerTokenRef.set(customerToken);
         this.callbackScheduler = callbackScheduler;
     }
 
     @Override
-    public void setCustomerToken(CustomerToken customerToken) {
-        this.customerToken = customerToken;
-    }
-
-    @Override
     public CustomerToken getCustomerToken() {
-        return customerToken;
+        final CustomerToken customerToken = customerTokenRef.get();
+        if (customerToken != null) {
+            return new CustomerToken(customerToken);
```

Comment:
agreed, I don't think that it is bad to protect mutability, but CustomerToken has only private fields, and no setters so is immutable in practice right now.  Is this protection against future changes in CustomerToken?